test/files/rules/no-duplicate-variable.test.ts

```diff
@@ -151,4 +151,9 @@ function testDestructuring() {
 
     var [a2, [b2, c2]] = [1, [2, 3]];
     var [{a2, d2}] = [{a2: 1, d2: 4}]; // failure
+
+    function myFunc2([a, b]) {
+        var a; // not a failure; caught by no-shadowed-variable
```

Comment:
hmm it looks like no-duplicate-variable has some issues in its current state... let's take that up in a separate PR since this one is already getting out of scope.

for example, line 98 in this file really ought to report a failure (and there's no such test for that syntax for `no-shadowed-variable`).